ifrs17/Report/ReportMutableScopes.ipynb

```diff
@@ -293,11 +293,13 @@
                 "\n                                        string reportType, (string filterName, object filterValue)[] dataFilter)> ",
                 "\n{",
                 "\n    IDataCube<ReportVariable> Cube { get{",
-                "\n        var dataCube = GetScope<IIfrs17Report>(Identity.reportType).GetDataCube();",
+                "\n        var data = GetScope<IIfrs17Report>(Identity.reportType).GetDataCube();",
                 "\n        // TODO: suggestion to place the filter here instead of having it in every applicability scope",
-                "\n        if(Identity.scenario != \"Delta\") return dataCube;",
-                "\n        var bestEstimateById = dataCube.Where(x => x.Scenario == null).ToDictionary(x => x.ToIdentityString());",
-                "\n        return dataCube.Select(x => x.Scenario == null ? x : x with { Value = x.Value - (bestEstimateById.TryGetValue((x with {Scenario = null}).ToIdentityString(), out var be)? be.Value : 0.0) }).ToDataCube();",
+                "\n        if(Identity.scenario != ScenariosAll && Identity.scenario != ScenariosDelta) return data;",
+                "\n        if(Identity.scenario == ScenariosAll) return data.Select(x => x.Scenario == null? x with {Scenario = ScenarioDefault } : x).ToDataCube();",
+                "\n        var bestEstimateById = data.Where(x => x.Scenario == null).ToDictionary(x => x.ToIdentityString());",
```

Comment:
Consider using ToStringWith instead of ToIdentityString to increase performance. Be aware that different clients might have different IP for reportVariables